test/shared/src/main/scala/zio/test/package.scala

```diff
@@ -43,8 +44,34 @@ import zio.stream.{ ZSink, ZStream }
  * }}}
  */
 package object test extends CompileVariants {
+
   type AssertResult = BoolAlgebraM[Any, Nothing, AssertionValue]
-  type TestResult   = BoolAlgebraM[Any, Nothing, FailureDetails]
+
+  /**
+   * A `PerTestAtLeastR[R]` is a `TestAspect.PerTest` that that requires at least an R in its environment
+   */
+  type PerTestAtLeastR[R] =
```

Comment:
how about my last commit? It's only used in construction of test aspects; all return types are are still `TestAspect`s, but saves a lot of `[Nothing, Any, Nothing, Any, Nothing, Any]` clogging up the file and locates it with the `PerTest` trait. I think it reads much nicer like this rather than ditching them altogether.